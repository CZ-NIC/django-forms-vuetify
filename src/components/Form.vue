<template>
    <form :action="send_url" :method="send_method">
        <slot name="form-errors">
            <ul v-if="source_form.form_errors">
                <li v-for="(error, index) in source_form.form_errors" :key="index">
                    {{ error }}
                </li>
            </ul>
        </slot>
        <slot name="autogenerated-items">
            <div v-if="source_form.fields">
                <div v-for="(field, key) in source_form.fields" :key="key">
                    <FormItem :ref="field.name" :field="field"/>
                </div>
            </div>
        </slot>
        <slot name="additional-items"/>
        <v-btn type="submit" primary color="primary">
            Send
        </v-btn>
    </form>
</template>

<script lang="ts">
import 'reflect-metadata'
import { Vue, Component, Prop } from 'vue-property-decorator'
import FormItem, { FieldInterface } from '@/components/FormItem.vue'

interface FormInterface {
    form_errors: string[];
    fields: FieldInterface[];
}

@Component({
    components: {
        FormItem,
    },
})
export default class Form extends Vue {
    @Prop({default: () => ({}) })
    readonly source_form!: FormInterface
    @Prop({required: true})
    readonly send_url!: string
    @Prop({default: 'get'})
    readonly send_method!: string
}
</script>
