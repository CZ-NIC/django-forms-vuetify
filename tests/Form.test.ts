import { shallowMount, Wrapper } from '@vue/test-utils'
import Form from '@/components/Form.vue'

describe('Form snapshot', () => {

    let wrapper: Wrapper<Form>

    beforeEach(() => {
        wrapper = shallowMount(Form, {
            propsData: {
                source_form: {
                    form_errors: ['Test error 1', 'Test error 2'],
                    fields: [{
                        name: 'test_input',
                        label: 'Test label',
                        help_text: 'Help for test input',
                        value: 'test value',
                        required: true,
                        errors: ['Error'],
                        widget: {name: 'input', type: 'text'}
                    }],
                },
                send_method: 'post',
                send_url: 'test.local/send',
            },
        })
    })

    test('matches snapshot', () => {
        expect(wrapper).toMatchSnapshot()
    })
})
