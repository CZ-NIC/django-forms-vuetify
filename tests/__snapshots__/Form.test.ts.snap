// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form snapshot matches snapshot 1`] = `
<form action="test.local/send" method="post">
  <ul>
    <li>
      Test error 1
    </li>
    <li>
      Test error 2
    </li>
  </ul>
  <div>
    <div>
      <formitem-stub field="[object Object]" value="test value"></formitem-stub>
    </div>
  </div>
  <v-btn-stub color="primary" tag="button" activeclass="" type="submit" primary="">
    Send
  </v-btn-stub>
</form>
`;
